ListManagement/processOpenSlidesParticipants.py

```diff
@@ -0,0 +1,184 @@
+
+    def toOpenSlidesRow(self):
+        return OpenSlidesRow(firstName=self.firstName, lastName=self.lastName, email=self.email, pronoun=self.pronoun)
+
+def readInANList(path:str ) -> list[ANReportRow]:
+    result = []
+    with open(file=path, newline='') as csvFile:
+        reader = csv.DictReader(csvFile)
+        for row in reader:
+            result.append(ANReportRow(firstName=row[Constants.ANCols.FIRST_NAME], 
+                                      lastName=row[Constants.ANCols.LAST_NAME], 
+                                      email=row[Constants.ANCols.EMAIL],
+                                      pronoun=row[Constants.ANCols.PRONOUNS]))
+    return result
+
+def readInOpenSlidesList(path:str ) -> list[OpenSlidesRow]:
+    result = []
+    with open(file=path, newline='') as csvFile:
+        reader = csv.DictReader(csvFile)
+        for row in reader:
+            # We want to ignore the isActive, isPresent, isPhysicalPerson, voteWeight, number, comment, and groups columns
+            result.append(OpenSlidesRow(title=row[Constants.OpenSlidesCols.TITLE],
+                                        firstName=row[Constants.OpenSlidesCols.FIRST_NAME], 
+                                        lastName=row[Constants.OpenSlidesCols.LAST_NAME], 
+                                        email=row[Constants.OpenSlidesCols.EMAIL],
+                                        pronoun=row[Constants.OpenSlidesCols.PRONOUN],
+                                        gender=row[Constants.OpenSlidesCols.GENDER],
+                                        username=row[Constants.OpenSlidesCols.USERNAME],
+                                        defaultPassword=row[Constants.OpenSlidesCols.DEFAULT_PASSWORD],
+                                        saml_id=row[Constants.OpenSlidesCols.SAML_ID],
+                                        structreLevel=row[Constants.OpenSlidesCols.STRUCTRE_LEVEL]))
+    return result
+
+def main(args):
+    setup()
+    flags = parseArgs()
+    logging.info("Reading Open Slides Previous list, %s", flags.prevListPath)
+    previousList = readInOpenSlidesList(flags.prevListPath)
+    logging.info("Reading in AN list, %s", flags.mebershipListPath)
+    anList = readInANList(flags.mebershipListPath)
+
+    outputList: list[OpenSlidesRow] = []
+
+    # Loop through AN list
+    # If a user with the same email exists in the previous list, add that row to the output
+    # Otherwise create a new row for that user
+    for anRow in anList:
+        logging.info("Looking for %s", anRow.email)
+        rowToAdd = None
+        for oldRow in previousList:
+            if anRow.email.casefold() == oldRow.email.casefold():
+                rowToAdd = oldRow
+                logging.info("Found info for %s", anRow.email)
+                break
+        if rowToAdd is None:
+            logging.info("Nothing found for %s, making new row", anRow.email)
+            rowToAdd = anRow.toOpenSlidesRow()
+        outputList.append(rowToAdd)
```

Comment:
Need an `else` or something here, otherwise the final list is going to have a lot of `None`s in it
```suggestion
        else:
          outputList.append(rowToAdd)
```